pkg/vpcmodel/grouping.go

```diff
@@ -214,52 +228,83 @@ func subnetGrouping(groupedConnLines *GroupConnLines,
+	if vsi {
+		allowedConnsCombined = g.v.AllowedConnsCombined.nodesConnectivityToGeneralConnectivity()
+	} else {
+		allowedConnsCombined = g.s.AllowedConnsCombined
+	}
 	res := []*GroupedConnLine{}
-	for src, nodeConns := range g.v.AllowedConnsCombined {
+	for src, nodeConns := range allowedConnsCombined {
 		for dst, conns := range nodeConns {
-			if conns.IsEmpty() {
-				continue
+			err := g.addLineToExternalGrouping(&res, conns.IsEmpty(), src, dst, conns.EnhancedString())
+			if err != nil {
+				return err
 			}
-			connString := conns.EnhancedString()
-			switch {
-			case dst.IsPublicInternet():
-				g.srcToDst.addPublicConnectivity(src, connString, dst)
-			case src.IsPublicInternet():
-				g.dstToSrc.addPublicConnectivity(dst, connString, src)
-			default:
-				res = append(res, &GroupedConnLine{src, dst, connString})
+		}
+	}
+	g.appendGrouped(&res)
+	return nil
+}
+
+func (g *GroupConnLines) groupExternalAddressesForDiff(thisMinusOther bool) error {
+	// initialize data structures
+	g.srcToDst = newGroupingConnections()
+	g.dstToSrc = newGroupingConnections()
+	var res []*GroupedConnLine
+	var connRemovedChanged connectivityDiff
+	if thisMinusOther {
+		connRemovedChanged = g.d.cfg1ConnRemovedFrom2
+	} else {
+		connRemovedChanged = g.d.cfg2ConnRemovedFrom1
+	}
+	for src, endpointConnDiff := range connRemovedChanged {
+		for dst, connDiff := range endpointConnDiff {
+			connDiffString := connDiffEncode(src, dst, connDiff, thisMinusOther)
+			connsEmpty := connDiff.conn1.IsEmpty() && connDiff.conn2.IsEmpty()
+			err := g.addLineToExternalGrouping(&res, connsEmpty, src, dst, connDiffString)
+			if err != nil {
+				return err
 			}
 		}
 	}
-	// add to res lines from  srcToDst and DstToSrc groupings
-	res = append(res, g.srcToDst.getGroupedConnLines(g, true)...)
-	res = append(res, g.dstToSrc.getGroupedConnLines(g, false)...)
-	g.GroupedLines = res
+	g.appendGrouped(&res)
+	return nil
 }
 
-func (g *GroupConnLines) groupExternalAddressesForSubnets() {
-	// groups public internet ranges in dst when dst is public internet
-	res := []*GroupedConnLine{}
-	for src, endpointConns := range g.s.AllowedConnsCombined {
-		for dst, conns := range endpointConns {
-			if conns.IsEmpty() {
-				continue
-			}
-			connString := conns.EnhancedString()
-			if dstNode, ok := dst.(Node); ok && dstNode.IsPublicInternet() {
-				g.srcToDst.addPublicConnectivity(src, connString, dstNode)
-			} else { // since pgw enable only egress src can not be public internet, the above is the only option of public internet
-				// not an external connection in source or destination - nothing to group, just append
-				res = append(res, &GroupedConnLine{src, dst, connString})
-			}
-		}
+func (g *GroupConnLines) addLineToExternalGrouping(res *[]*GroupedConnLine, emptyConn bool,
+	src, dst VPCResourceIntf, connEnhanced string) error {
+	if emptyConn {
+		return nil
+	}
+	srcNode, srcIsNode := src.(Node)
+	dstNose, dstIsNode := dst.(Node)
```

Comment:
```suggestion
	dstNode, dstIsNode := dst.(Node)
```

Reply:
done - not accepting the suggestion since it does not include all occurrences